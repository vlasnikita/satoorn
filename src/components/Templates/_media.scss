@media only screen
and (max-width : 1300px){
  .Templates{
    width: 370px;
    min-width: 370px;

    &__template{
      height: 18px;
      line-height: 18px;
      margin-bottom: 8px;
      margin-right: 8px;
    }

    &__title{
      font-size: 1.4rem;
    }
  }
}

@media only screen
and (max-width : 900px){
  .Templates{
    width: 260px;
    min-width: 260px;

    &__header {
      font-size: 1.8rem;
      height: 40px;
      line-height: 40px;
      margin-left: 10px;
    }

    &__feed{
      width: calc(100% - 20px);
      padding: 10px;
    }

    &__template {
      height: 14px;
      line-height: 14px;
      margin-bottom: 6px;
      margin-right: 8px;
      border-radius: 5px;
      padding: 3px 6px;
    }

    &__title {
      font-size: 1.2rem;
      font-weight: 400;
    }
  }
}

@media only screen
and (max-width : 650px){
  .Templates{
    display: none;
  }
}