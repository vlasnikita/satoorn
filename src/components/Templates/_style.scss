@mixin multiLineEllipsis($lineHeight: 1.3rem, $lineCount: 3) {
  overflow: hidden;
  position: relative;
  line-height: $lineHeight;
  max-height: $lineHeight * $lineCount;
}

.Templates {
  background: #282828;
  width: 550px;
  min-width: 550px;
  height: 100%;
  transition: 0s;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  overflow-y: scroll;

  &__header{
    color: #bbb;
    font-size: 2.2rem;
    font-weight: normal;
    margin-right: auto;
    height: 60px;
    line-height: 60px;
    margin-left: 20px;
  }

  &__feed{
    height: auto;
    width: calc(100% - 40px);
    padding: 20px;
    padding-top: 0;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: flex-start;
    position: relative;
  }

  &__template {
    border-radius: 8px;
    position: relative;
    height: 26px;
    line-height: 26px;
    padding: 5px 10px;
    width: auto;
    margin-bottom: 12px;
    margin-right: 12px;
    background: #fff;
    cursor: pointer;
    overflow: hidden;
    animation: TransitioningBackground 10s ease infinite;
    transition: 1s;

    &::before {
      content: '';
      display: block;
      position: absolute;
      background: rgba(255, 255, 255, 0.5);
      width: 60px;
      height: 100%;
      top: 0;
      filter: blur(30px);
      transform: translateX(-100px) skewX(-15deg);
    }

    &::after {
      content: '';
      display: block;
      position: absolute;
      background: rgba(255, 255, 255, 0.2);
      width: 30px;
      height: 100%;
      top: 0;
      filter: blur(5px);
      transform: translateX(-100px) skewX(-15deg);
    }

    &:hover{
      transition: 0s;
      background: #eee;
    }

    &_highlighted {
      background-color: #7d72cf;
      background-image: (linear-gradient(to left, #2d8fe5, #d155b8));

      .Templates__title{
        color: #eee;
        transition: 0s;
      }

      &::before,
      &::after {
        transform: translateX(500px) skewX(-15deg);
        transition: 3.5s;
      }
    }
  }

  &__image {
    width: 130px;
    position: relative;

    &Container{
      position: relative;
      width: 130px;
      height: auto;
      margin-right: 10px;
      margin-bottom: 5px;
      float: left;
    }
  }

  &__body {
    padding: 0 5px;
    position: absolute;
    bottom: 0;
    left: 0;
    width: calc(100% - 10px);
    height: 60%;
    padding-top: 10%;
  }

  &__title {
    font-weight: bold;
    font-size: 1.6rem;
    color: #222;
    transition: 1s;
  }

  &__description {
    font-size: 1.2rem;
    margin-top: 10px;
    color: #222;
  }

  &__copy {
    z-index: 100;
    height: 100%;
    width: 100%;
    background: rgba(34, 34, 34, .97);
    position: absolute;
    top: 0;
    left: 0;
    outline: none;
    border: none;
    border-radius: 4px;
    display: none;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    &:active{
      .Templates__copyText{
        border: 2px solid $mainYellow;
      }
    }

    &Text {
      font-weight: bold;
      font-size: 1.4rem;
      color: #eeeeee;
      display: flex;
      justify-content: center;
      align-items: center;
      width: 120px;
      height: 120px;
      border: 2px dashed #eee;
      border-radius: 50%;
      background: #333;
    }
  }

  &__button{
    border: none;
    text-decoration: underline;
    outline: none;
    margin-top: 5px;
    padding: 0;
    text-align: right;
    padding-left: 5px;
    width: 100%;
    font-size: 1.2rem;
    color: #666;
    font-style: italic;
  }
}



@keyframes TransitioningBackground {
  0% {
    background-position: 1% 0%;
  }
  50% {
    background-position: 99% 100%;
  }
  100% {
    background-position: 1% 0%;
  }
}

@import "/media";