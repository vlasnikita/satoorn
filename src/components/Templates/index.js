import React, {Component} from 'react';
import {connect} from 'react-redux';

import {setCreateChallengeDescription, setCreateChallengeType} from "Actions/createChallenge";
import pipe from 'Static/templates/templates-pipe.svg'

let lastHighlightedIndex = 0

class Templates extends Component {

    componentDidMount() {
        const templatesNodes = document.querySelectorAll('.Templates__template')

        setInterval(()=>{

            let indexToHighlight = Math.floor(Math.random() * templatesNodes.length - 1)
            indexToHighlight = indexToHighlight === lastHighlightedIndex ? Math.floor(Math.random() * templatesNodes.length - 1) : indexToHighlight

            const highlightedTemplate = templatesNodes[indexToHighlight]
            highlightedTemplate.className = 'Templates__template Templates__template_highlighted'

            lastHighlightedIndex = indexToHighlight
            setTimeout(() => highlightedTemplate.className = 'Templates__template',1100)

        }, 5000)
    }

    render() {
        return (
            <div className="Templates">
                <h3 className="Templates__header">Рекомендации</h3>
                <div className="Templates__feed">
                    {this.props.templates.map((template, i) => (
                        <div
                            key={i}
                            className="Templates__template"
                            onClick={() => this.props.handleSetCreateChallengeDescription(template.description)}
                        >
                            <p className="Templates__title">{template.title}</p>
                        </div>
                    ))}
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    templates: state.templates.templates
});

const mapDispatchToProps = (dispatch) => ({
    handleSetCreateChallengeDescription: description => dispatch(setCreateChallengeDescription(description)),
    handleSetCreateChallengeType: type => dispatch(setCreateChallengeType(type))
})

export default connect(mapStateToProps, mapDispatchToProps)(Templates);