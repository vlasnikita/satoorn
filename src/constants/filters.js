export const TEMPLATE_FILTERS = [
    {
        key: 'gaming',
        value: 'Игровые'
    },
    {
        key: 'other',
        value: 'Неигровые'
    }
]

export const CHALLENGE_FILTERS = [
    {
        key: 'created',
        value: 'Доступные'
    },
    {
        key: 'accepted',
        value: 'Принятые'
    }
]
