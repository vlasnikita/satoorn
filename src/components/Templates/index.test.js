import React from 'react'
import 'jest-specific-snapshot'
import renderer from 'react-test-renderer'
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter } from 'react-router-dom';

import Templates from './index'
import {unmountComponentAtNode} from "react-dom";
import {TEMPLATE_FILTERS} from "Constants/filters";

const mockStore = configureMockStore();

const allTemplates = require('Mocks/templates')
const initialState = {
    templates: {
        allTemplates,
        templates: allTemplates.filter(template => template.type === TEMPLATE_FILTERS[0].key),
        filter: TEMPLATE_FILTERS[0]
    }
}

it('renders correctly', () => {
    function createNodeMock(element) {
        if (element.className === 'Challenge__description') {
            // This is your fake DOM node for <p>.
            // Feel free to add any stub methods, e.g. focus() or any
            // other methods necessary to prevent crashes in your components.
            return {};
        }
        // You can return any object from this method for any type of DOM component.
        // React will use it as a ref instead of a DOM node when snapshot testing.
        return element
    }

    const store = mockStore(initialState);
    const options = {createNodeMock}
    const tree = renderer.create(
        <BrowserRouter>
            <Provider store={store}>
                <Templates
                />
            </Provider>
        </BrowserRouter>,
        options
    )

    expect(tree.toJSON()).toMatchSnapshot();
});