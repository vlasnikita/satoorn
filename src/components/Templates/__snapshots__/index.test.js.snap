// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Templates"
>
  <h3
    className="Templates__header"
  >
    Рекомендации
  </h3>
  <div
    className="Templates__feed"
  >
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 3
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 5
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 7
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 799
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 8
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 800
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 805
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 807
      </p>
    </div>
    <div
      className="Templates__template"
      onClick={[Function]}
    >
      <p
        className="Templates__title"
      >
        It's a challenge template with number 999
      </p>
    </div>
  </div>
</div>
`;
